File: functional01-java/src/main/java/org/example/vavr/login/UserDAO.java
package org.example.vavr.login;

import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class UserDAO {
    // it simulates a DB with 10 users
    private List<User> users = IntStream.rangeClosed(1, 10)
            .mapToObj(i -> new User("user" + i, "pass" + i))
            .collect(Collectors.toList());
    public List<User> getUsers() {
        return users;
    }
}

File: functional-js/proj02/loginController/userDao.js
// Define the User class
class User {
    constructor(username, password) {
        this.username = username;
        this.password = password;
    }
}

// Define the UserDAO class
class UserDAO {
    constructor() {
        // Simulate a DB with 10 users
        this.users = Array.from({ length: 10 }, (_, i) => new User(`user${i + 1}`, `pass${i + 1}`));
    }

    getUsers() {
        return this.users;
    }
}

module.exports = { UserDAO }

// // Example usage
// const userDao = new UserDAO();
// const users = userDao.getUsers();
// console.log(users);
